### ISSUE TITLE
Kubernetes Mercure hub publish performance

### ISSUE THREAD

**Issue by mkurzeja** [dunglas/mercure#703]
TLDR: Mercure Hub on K8s is 3x slower than one run locally in Docker, when using 500 virtual users. 

We are running an open-source version of Mercure Hub on Kubernetes - installed using the official Helm chart.

values.yaml:
```
publisherJwtKey: !ChangeMe!
subscriberJwtKey: !ChangeMe!
persistence:
  enabled: false
```

During load tests of our system, using 500 simultaneous users, we noticed that publishing a message takes a very long time. An average of almost 2s, and a plateau of the concurrent connections it can handle when publishing.  The more we throw users, the longer it takes to respond. The CPU/RAM usage is rather low on the Mercure hub.

What I noticed, when we raise the VUS (users) too high, **even the liveness probe fails.** 

So what I did today, was to write a simple k6 script to test publishing only:

```
import http from 'k6/http';

export const options = {
    vus: 500,
    insecureSkipTLSVerify: true,
    duration: '60s',
  };

export default function () {
    const url = 'http://mercure.production.svc.cluster.local/.well-known/mercure';
    const payload = {
      topic: 'https://example.com/books/1',
      data: '{"foo": "updated value"}'
    };
  
    const params = {
      headers: {
        'Authorization': 'Bearer ...',
      },
    };
  
    http.post(url, payload, params);

}
```

And run it, the average response was 1.4s:

<img width="806" alt="image" src="https://user-images.githubusercontent.com/1044032/194720685-a19da894-7017-468a-b9e2-a69443946815.png">

(this is one of 4 workers running on the cluster and emulating the load).

This is strange, as the same test run locally against my Docker-run Mercure hub is 3 times faster:

<img width="916" alt="image" src="https://user-images.githubusercontent.com/1044032/194720673-d3c57f31-be81-4682-87b2-11adc5cc3bd7.png">

I did some research and noticed that ulimit might be the issue, not sure if I'm checking it correctly, but it seems not to be the case for us:
<img width="502" alt="image" src="https://user-images.githubusercontent.com/1044032/194720718-91eb99cb-b646-45c4-83c0-21ceb18afb2d.png">

As the load test is using direct access using a service, I have no idea what could be limiting the number of concurrent connections.

Any ideas or suggestions?

**Comment by mkurzeja** [comment]
Sorry for missing this in the initial description, but thanks a lot for Mercure Hub. Does a great job for us! 

Let me know if you would like to run some more tests with some changes.

**Comment by mkurzeja** [comment]
Ok, some more tests, and interesting outcomes:

1. Run same scenario on a local (docker-desktop) k8s cluster - performance exactly the same as local docker env == fast
2. Bought a small scaleway k8s cluster - performance exactly the same as on Azure k8s. 
3. On "slow" k8s clusters, with 60s the number of requests handled is the same no matter if I run 500vus or 1000vus. 
4. The long duration is visible in mercure output (1000vus):

```
1.6653395325552833e+09	info	http.log.access.log0	handled request	{"request": {"remote_ip": "100.64.0.164", "remote_port": "49282", "proto": "HTTP/1.1", "method": "POST", "host": "xxx.nip.io", "uri": "/.well-known/mercure", "headers": {"Authorization": [], "X-Forwarded-Host": ["xxx.nip.io"], "X-Forwarded-Proto": ["http"], "User-Agent": ["k6/0.40.0 (https://k6.io/)"], "Content-Length": ["87"], "Content-Type": ["application/x-www-form-urlencoded"], "X-Forwarded-For": ["x.x.x.x"], "X-Forwarded-Port": ["80"], "X-Forwarded-Server": ["traefik-5b78dd79dc-zrv8g"], "X-Real-Ip": ["x.x.x.x"], "Accept-Encoding": ["gzip"]}}, "user_id": "", "duration": 3.503573124, "size": 45, "status": 200, "resp_headers": {"Content-Security-Policy": ["default-src 'self'"], "Server": ["Caddy"], "X-Frame-Options": ["DENY"], "X-Content-Type-Options": ["nosniff"], "X-Xss-Protection": ["1; mode=block"]}}
```
> "duration": 3.503573124,

On this stage I would assume this is a network limit on my side, but the duration was the same when I run the tests from the k8s cluster, plus Mercure clearly notices the duration. Maybe this is IO?

**Comment by mkurzeja** [comment]
Ok, long story short: this is the boltdb being able to write only from one request at a time, and this is limited by the IO of the disk. Switching to `local` transport improved the performance a lot.
